presto-bigquery/src/test/java/com/facebook/presto/plugin/bigquery/TestTypeConversions.java

```diff
@@ -0,0 +1,302 @@
+import static java.time.Month.FEBRUARY;
+import static java.time.Month.JANUARY;
+import static java.time.Month.JUNE;
+import static java.time.Month.MARCH;
+import static java.time.Month.MAY;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Test
+public class TestTypeConversions
+{
+    @Test
+    public void testConvertBooleanField()
+    {
+        assertSimpleFieldTypeConversion(LegacySQLTypeName.BOOLEAN, BooleanType.BOOLEAN);
+    }
+
+    @Test
+    public void testConvertBytesField()
+    {
+        assertSimpleFieldTypeConversion(LegacySQLTypeName.BYTES, VarbinaryType.VARBINARY);
+    }
+
+    @Test
+    public void testConvertDateField()
+    {
+        assertSimpleFieldTypeConversion(LegacySQLTypeName.DATE, DateType.DATE);
+    }
+
+    @Test
+    public void testConvertDateTimeField()
+    {
+        assertSimpleFieldTypeConversion(LegacySQLTypeName.DATETIME, TimestampType.TIMESTAMP);
+    }
+
+    @Test
+    public void testConvertFloatField()
+    {
+        assertSimpleFieldTypeConversion(LegacySQLTypeName.FLOAT, DoubleType.DOUBLE);
+    }
+
+    @Test
+    public void testConvertGeographyField()
+    {
+        assertSimpleFieldTypeConversion(LegacySQLTypeName.GEOGRAPHY, VarcharType.VARCHAR);
+    }
+
+    @Test
+    public void testConvertIntegerField()
+    {
+        assertSimpleFieldTypeConversion(LegacySQLTypeName.INTEGER, BigintType.BIGINT);
+    }
+
+    @Test
+    public void testConvertNumericField()
+    {
+        assertSimpleFieldTypeConversion(LegacySQLTypeName.NUMERIC, DecimalType.createDecimalType(38, 9));
+    }
+
+    @Test
+    public void testConvertStringField()
+    {
+        assertSimpleFieldTypeConversion(LegacySQLTypeName.STRING, VarcharType.VARCHAR);
+    }
+
+    @Test
+    public void testConvertTimeField()
+    {
+        assertSimpleFieldTypeConversion(LegacySQLTypeName.TIME, TimeWithTimeZoneType.TIME_WITH_TIME_ZONE);
+    }
+
+    @Test
+    public void testConvertTimestampField()
+    {
+        assertSimpleFieldTypeConversion(LegacySQLTypeName.TIMESTAMP, TimestampWithTimeZoneType.TIMESTAMP_WITH_TIME_ZONE);
+    }
+
+    @Test
+    public void testConvertOneLevelRecordField()
+    {
+        Field field = Field.of(
+                "rec",
+                LegacySQLTypeName.RECORD,
+                Field.of("sub_s", LegacySQLTypeName.STRING),
+                Field.of("sub_i", LegacySQLTypeName.INTEGER));
+        ColumnMetadata metadata = Conversions.toColumnMetadata(field);
+        RowType targetType = RowType.from(ImmutableList.of(
+                RowType.field("sub_s", VarcharType.VARCHAR),
+                RowType.field("sub_i", BigintType.BIGINT)));
+        assertThat(metadata.getType()).isEqualTo(targetType);
+    }
+
+    @Test
+    public void testConvertTwoLevelsRecordField()
+    {
+        Field field = Field.of(
+                "rec",
+                LegacySQLTypeName.RECORD,
+                Field.of("sub_rec", LegacySQLTypeName.RECORD,
```

Comment:
could we static import LegacySQLTypeName.RECORD, LegacySQLTypeName.STRING, and all other names?

Reply:
unable because the type for imported variable is now `bigquery.LegacySQLTypeName` which does not match with required type which is `LegacySQLTypeName`

Reply:
do not understand, could we do:
static import com.google.cloud.bigquery.LegacySQLTypeName.RECORD;

and use RECORD directly?

Reply:
right, got it, fixed